docs/platforms/react-native/integrations/error-boundary.mdx

```diff
@@ -0,0 +1,194 @@
+
+`onError` (Function)
+
+A function that gets called when the Error Boundary encounters an error. `onError` is useful if you want to propagate the error into a state management library like Redux, or if you want to check any side effects that could have occurred due to the error.
+
+`onMount` (Function)
+
+A function that gets called on ErrorBoundary `componentDidMount()`.
+
+`onUnmount` (Function)
+
+A function that gets called on ErrorBoundary `componentWillUnmount()`.
+
+`beforeCapture` (Function)
+
+A function that gets called before an error is sent to Sentry, allowing for extra tags or context to be added to the error.
+
+## Examples
+
+#### Setting a Fallback Function (Render Props)
+
+Below is an example where a fallback prop, using the [render props approach](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop), is used to display a fallback UI on error. The fallback UI returns to a standard component state when reset using the `resetError()` API provided by the component through render props.
+
+```javascript
+import React from "react";
+import { Button, Text } from "react-native"
+import * as Sentry from "@sentry/react-native";
+
+class MyComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      message: "This is my app",
+    };
+  }
+
+  render() {
+    return (
+      <Sentry.ErrorBoundary
+        fallback={({ error, componentStack, resetError }) => (
+          <React.Fragment>
+            <Text>You have encountered an error</Text>
+            <Text>{error.toString()}</Text>
+            <Text>{componentStack}</Text>
+            <Button
+              title="Click here to reset!"
+              onPress={() => {
+                this.setState({ message: "This is my app" });
+                // When resetError() is called it will
+                // remove the Fallback component and render
+                // the Sentry ErrorBoundary's children
+                // in their initial state
+                resetError();
+              }}
+            />
+          </React.Fragment>
+        )}
+      >
+        <Text>{this.state.message}</Text>
+        {/* on click, this button sets an Object */}
+        {/* as a message, not a string. Which will cause */}
+        {/* an error to occur in the component tree */}
+        <Button
+          title="Click here to change message!"
+          onPress={() => this.setState({ message: { text: "Hello World" } })}
+        />
+      </Sentry.ErrorBoundary>
+    );
+  }
+}
+
+export default MyComponent;
+```
+
+#### Using multiple error boundaries
+
+When using multiple error boundaries, we recommend using `beforeCapture` to set tags/context so that you can tell which error boundary the error occurred from. In the example below, we attach tags to errors based on what route they rendered in.
```

Comment:
```suggestion
When using multiple error boundaries, we recommend using `beforeCapture` to set tags/context so that you can tell which error boundary the error occurred from. In the example below, we attach tags to errors based on what route they rendered in:
```